src/CabanaPD_Solver.hpp

```diff
@@ -580,6 +568,51 @@ class SolverFracture
         this->final_output();
     }
 
+    void run()
+    {
+        this->init_output( 0.0 );
+
+        // Main timestep loop.
+        for ( int step = 1; step <= num_steps; step++ )
+        {
+            _step_timer.start();
+
+            // Integrate - velocity Verlet first half.
+            integrator->initialHalfStep( *particles );
+
+            if constexpr ( std::is_same<typename force_model_type::thermal_type,
+                                        TemperatureDependent>::value )
+                comm->gatherTemperature();
+
+            // Update ghost particles.
+            comm->gatherDisplacement();
+
+            // Update force and related quantities.
+            updateForce();
+
+            // Integrate - velocity Verlet second half.
+            integrator->finalHalfStep( *particles );
+
+            output( step );
+        }
+
+        // Final output and timings.
+        this->final_output();
+    }
+
+    void updateForce()
```

Comment:
Why do we have "updateForce" here and also in Line 306?

Reply:
This is a different class